cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/AbstractProgramRuntimeService.java

```diff
@@ -300,22 +300,19 @@ public boolean apply(RuntimeInfo info) {
   }
 
   @Override
-  public boolean checkAnyRunning(Predicate<Id.Program> predicate, ProgramType... types) {
+  public List<RuntimeInfo> listAll(ProgramType... types) {
+    List<RuntimeInfo> runningPrograms = new ArrayList<>();
     for (ProgramType type : types) {
-      for (Map.Entry<RunId, ProgramRuntimeService.RuntimeInfo> entry :  list(type).entrySet()) {
+      for (Map.Entry<RunId, ProgramRuntimeService.RuntimeInfo> entry : list(type).entrySet()) {
         ProgramController.State programState = entry.getValue().getController().getState();
         if (programState.isDone()) {
           continue;
         }
-        Id.Program programId = entry.getValue().getProgramId();
-        if (predicate.apply(programId)) {
-          LOG.trace("Program still running in checkAnyRunning: {} {} {} {}",
-                    programId.getApplicationId(), type, programId.getId(), entry.getValue().getController().getRunId());
-          return true;
-        }
+        RuntimeInfo programId = entry.getValue();
+        runningPrograms.add(programId);
```

Comment:
The variable shouldn't be called `programId` anymore. Also, you can simplify it to `runningPrograms.add(entry.getValue())`.